src/core/operations/RAKE.mjs

```diff
@@ -0,0 +1,149 @@
+            }
+        ];
+    }
+
+    /**
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {string}
+     */
+    run(input, args) {
+
+        // Get delimiter regexs
+        const wordDelim =  new RegExp(args[0], "g");
+        const sentDelim = new RegExp(args[1], "g");
+
+        // Deduplicate the stop words and add the empty string
+        const stopWords = args[2].toLowerCase().replace(/ /g, "").split(",").unique();
+        stopWords.push("");
+
+        // Lower case input and remove start and ending whitespace
+        input = input.toLowerCase().trim();
+
+        // Get tokens, token count, and phrases
+        const tokens = [];
+        const wordFrequencies = [];
+        let phrases = [];
+
+        // Build up list of phrases and token counts
+        const sentences = input.split(sentDelim);
+        for (const sent of sentences) {
+
+            // Split sentence into words
+            const splitSent = sent.split(wordDelim);
+            let startIndex = 0;
+
+            for (let i = 0; i < splitSent.length; i++) {
+                const token = splitSent[i];
+                if (stopWords.includes(token)) {
+                    // If token is stop word then split to create phrase
+                    phrases.push(splitSent.slice(startIndex, i));
+                    startIndex = i + 1;
+                } else {
+                    // If token is not a stop word add to the count of the list of words
+                    if (tokens.includes(token)) {
+                        wordFrequencies[tokens.indexOf(token)]+=1;
+                    } else {
+                        tokens.push(token);
+                        wordFrequencies.push(1);
+                    }
+                }
+            }
+            phrases.push(splitSent.slice(startIndex));
+        }
+
+        // remove empty phrases
+        phrases = phrases.filter(subArray => subArray.length > 0);
+
+        // Remove duplicate phrases
+        const uniquePhrases = [...new Set(phrases.map(function (phrase) {
+            return phrase.join(" ");
+        }))];
+        phrases = uniquePhrases.map(function (phrase) {
```

Comment:
Nitpick: In general I'd prefer arrow functions to this:

```javascript
phrases = uniquePhrases.map((phrase) => phrase.spit(' '))
```

Here, I'm not wrapping with curly braces so that it returns the statement automatically.  Note: Arrow functions don't bind `this`, which means arrow functions aren't automatically a replacement everywhere.

```javascript
function x() {
  y = () => {
    console.log(this) // refers to 'x', not 'y'!
  }

  function z() {
    console.log(this) // refers to 'z'
  }

  y()
  z()
}
```